widget/select_entry_test.go

```diff
@@ -1,4 +1,4 @@
-package widget_test
+package widget
```

Comment:
I'm very biased regarding this. Why did you move the tests into the `widget` package?

Reply:
All widgets and corresponding tests in this package has the same package name `widget`. So i don't see any reason of not maintaining the the same package name throughout all widget test. Moreover, it's better calling functions from the same package than making redundant imports. 

For the special cursorable button, we already discussed this on slack. SelectEntry dropdown button cursor is overriden by it's parent's (SelectEntry) cursor. This solves that issue.

Reply:
I consider testing a package via its exported API the better approach. The only exception from this are unit tests for complex internal methods. But these are rare in my experience.

So moving tests into the tested package just to be able to implement an unnecessary internal struct is a bad idea.